clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -312,31 +369,33 @@ public void testHeartbeatOutsideInterval() {
     @Test
     public void testNetworkTimeout() {
         // The initial heartbeatInterval is set to 0
-        resetWithZeroHeartbeatInterval(Optional.empty());
-        mockStableMember();
+        time.sleep(DEFAULT_HEARTBEAT_INTERVAL_MS);
         when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(new Node(1, "localhost", 9999)));
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(true);
         NetworkClientDelegate.PollResult result = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(1, result.unsentRequests.size());
         // Mimic network timeout
         result.unsentRequests.get(0).handler().onFailure(time.milliseconds(), new TimeoutException("timeout"));
 
+        time.sleep(1);
+        result = heartbeatRequestManager.poll(time.milliseconds());
+        assertEquals(1, result.unsentRequests.size());
+
         // Assure the manager will backoff on timeout
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.empty());
```

Comment:
this changes the semantics. This will achieve the result of not generating a HB, but we want to test that the HB is not generated because the backoff hasn't expired. So we should ensure there is a coordinator, ensure that the membershipMgr allows HB (shouldSKipHb false), and see that it's indeed the backoff check [here](https://github.com/apache/kafka/blob/a533e246e3ed5f6f6c5be4ebf9d29ae75cab557e/clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestState.java#L89) the one making that no HB is generated